README.md

```diff
@@ -1,2 +1,287 @@
+yarn add react-experimentify
+```
+
+### Optimize
+
+1. Decide on an experiment name
+
+* Guide `<Component under experiment>_<Placement Identifier>`
+
+  * eg: `h1_ListingPage`
+
+2. Set up an experiment in Google Optimize
+
+* Code to execute via Google Optimize
+
+  ```javascript
+  window.dispatchEvent(new CustomEvent('h1_ListingPage.render', {
+    detail: {
+      title: 'Treatment - ' + Date.now()
+    }
+  }));
+  ```
+* Add `h1_ListingPage.activate` as custom event
+
+  ![targeting](./docs/optimize_targeting.png)
+
+3. Import
+
+```javascript
+  import React from 'react';
+  import ReactDOM from 'react-dom';
+
+  import { Experiment, Treatment, Control, Optimize, ExperimentOpacifier } from 'react-experimentify';
+```
+
+4. Initialization
+
+```javascript
+import { Experiment, Treatment, Control, Optimize, ExperimentOpacifier } from 'react-experimentify';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+const experiment = new Optimize('h1_ListingPage');
+```
+
+5. Activation
+
+* Experiment can be activated at any point in time after component has been mounted. We can activate it on the
+client side as well. If we want the experiment to be activated straight after the parent component is mounted,
+add following
+
+```javascript
+componentDidMount() {
+  this.experiment.activate();
+  this.unsubscribe = this.experiment.subscribe(state => this.setState(state));
+}
+
+componentWillUnmount() {
+  this.unsubscribe();
+}
+```
+
+6. Add following to render method. _(Here we are experimenting _h1_ tag)_
+
+```javascript
+  <Experiment experiment={experiment}>
+    <div>
+      <Treatment>
+        <h1>Treatment 1</h1>
+      </Treatment>
+      <Control>
+        <p>Control</p>
+      </Control>
+    </div>
+  </Experiment>
+```
+
+7. With `ExperimentOpacifier`
+
+* Helps to reduce the flickering effect when the experiment is loading.
+* How it works?
+
+  * Let the component under experiment to render the control version in transparent mode.
+  * Wait for the experiment to kick in within the `timeout` period.
+  * If the experiment loads before timing out, user will see the variant of the experiment.
+  * If timed out, user will see control version.
+
+  ```javascript
+    <Experiment experiment={experiment}>
+      <ExperimentOpacifier
+        wrapper={'div'}
+        timeout={250}
+        transition={250}
+      >
+        <div>
+          <Treatment>
+            <h1>Treatment 1</h1>
+          </Treatment>
+          <Control>
+            <p>Control</p>
+          </Control>
+        </div>
+      </ExperimentOpacifier>
+    </Experiment>
+  ```
+
+8. Override opacity behaviour
+
+Because the `ExperimentOpacifier` initially renders with an opacity of `0`,
+this be override with the `.experiment__opacity` class.
+
+Add a `<noscript>` element to the page to override the opacity
+
+```html
+<noscript>
+  <style>
+    .experiment__opacity {
+      opacity: 1 !important;
+    }
+  </style>
+</noscript>
+```
+
+### Testing
```

Comment:
Because this is open source and only contributors would test this  - this belongs somewhere near the bottom of the readme